HHS/OPRE-OPS
# OPS-1555_Packages_and_Workflows

## What changed

- Introduced the Package system: for sending groups of items around for workflow needs; the initial implementation supports `BliPackages`
- Updated the `/agreements/review/{id}` to reflect this new package; instead of immediately updating the status of the BLI, it will now create a package; which is sent to review.
- Introduced the `Workflow` system; which allows us to create generic workflow templates, and individual workflow instances; 
  - each workflow contains a set of steps; those steps are the steps/tasks that the workflow requires
  - Example: An agreement needs two BLI sent for approval, A workflow instance is created, with a single `WorkflowStepInstance` that will require a anyone in a pre-defined set of people to review, then approve.
  - THIS IS ONLY THE BUILDING BLOCK FOR WORKFLOWS - NOT YET PUT TO USE.

## Issue

#1555

## How to test

1. Navigate to the Agreement page.
2. Select an agreement with DRAFT BLIs and click the "send for review" button (plane button?)
3. Select the `Change Draft Budget Lines` Action
4. Select the BLIs to test
5. Add a note
6. Click Send to Approval

Expected Results:
You should still see an alert that the BLIs were sent for review, and be sent back to the `/agreements` page.
There should now be a new record in the `BliPackages` and `Packages` tables


## Definition of Done Checklist


## Links

_If relevant, e.g. for a link to a piece of markdown documentation_
